test/autorest/generated/lrogroup/pollers.go

```diff
@@ -89,19 +60,22 @@ func (p *httpPoller) ResumeToken() (string, error) {
 	return string(js), nil
 }
 
-func httpPollerPollUntilDone(ctx context.Context, p HTTPPoller, frequency time.Duration) (*HTTPResponse, error) {
-	for !p.Done() {
+func httpPollerPollUntilDone(ctx context.Context, p HTTPPoller, frequency time.Duration) (*http.Response, error) {
```

Comment:
I just realized this should be a method on `httpPoller` as it takes a specific poller arg (my bad for not noticing this when I made the original suggestion).

```go
func (p *httpPoller) pollUntilDone(ctx context.Context, frequency time.Duration) (*http.Response, error)
```

It will require a slight change when creating the `PollUntilDone` func however; either a type assertion or capture the `httpPoller` in a local first (the latter might be better as we know the type and can avoid using reflection).

Reply:
So instead of having something like: 
`poller := result.Poller.(*httpPoller}` 
You're thinking of: 
`poller := &httpPoller{}`

Reply:
?

Reply:
Ignore my comment, I see what you mean now

Reply:
Done